### PR TITLE
feat: Add main thread ID to profiling payload

### PR DESCRIPTION
**Pull request by indragiek**
## :scroll: Description

Add a `is_main_thread` key to the thread metadata in the profiling payload that identifies the main thread of the application.

## :bulb: Motivation and Context

We currently rely on a potentially inaccurate heuristic on the backend to identify which thread is the main thread in an iOS profile. We can reliably get this info on the client instead of trying to guess on the backend.

## :green_heart: How did you test it?

Updated unit test.

## :pencil: Checklist

- [X] I reviewed the submitted code
- [X] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] Review from the native team if needed
- [X] No breaking changes

### CODE REVIEW

**Review by phacops — APPROVED** [review]

File: Tests/SentryTests/SentryHubTests.swift [hunk 1]
```diff
@@ -811,11 +811,12 @@ class SentryHubTests: XCTestCase {
...
         
         let sampledProfile = profile["sampled_profile"] as! [String: Any]
-        let threadMetadata = sampledProfile["thread_metadata"] as! [String: Any]
+        let threadMetadata = sampledProfile["thread_metadata"] as! [String: [String: Any]]
         let queueMetadata = sampledProfile["queue_metadata"] as! [String: Any]
         
         XCTAssertFalse(threadMetadata.isEmpty)
-        XCTAssertGreaterThan((threadMetadata.first?.value as! [String: Any])["priority"] as! Int, 0)
+        XCTAssertGreaterThan(threadMetadata.first?.value["priority"] as! Int, 0)
+        XCTAssertFalse(threadMetadata.values.filter { $0["is_main_thread"] as? Bool == true }.isEmpty)
```
Comment (bruno-garcia) [inline Tests/SentryTests/SentryHubTests.swift:819]: Can we have a test where the main thread shows up in the list?

**Review by bruno-garcia — APPROVED** [review]

File: Tests/SentryTests/SentryHubTests.swift (hunk 1 above)
Comment (indragiek) [reply Tests/SentryTests/SentryHubTests.swift:819]: `XCTAssertFalse(threadMetadata.values.filter { $0["is_main_thread"] as? Bool == true }.isEmpty)` is checking whether there is something in the list that is marked as being the main thread, did you mean something else?

**Review by brustolin — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## Unreleased
+
+### Features
+
+- Add main thread ID to profiling payload(#1918)
```
Comment (philipphofmann) [inline CHANGELOG.md:7]:
```suggestion
- Add main thread ID to profiling payload (#1918)
```

File: Sources/Sentry/SentryProfiler.mm [hunk 3]
```diff
@@ -93,12 +93,19 @@ @implementation SentryProfiler {
...
+    thread::TIDType _mainThreadID;
 }
 
 - (instancetype)init
 {
+    if (![NSThread isMainThread]) {
+        [SentryLog logWithMessage:@"SentryProfiler must be initialized on the main thread"
+                         andLevel:kSentryLevelError];
+        return nil;
+    }
```
Comment (philipphofmann) [inline Sources/Sentry/SentryProfiler.mm:105]: `m`: Can we add a test for this? Trying to init from a background thread and checking if the returned value is `nil`?

**Review by armcknight — APPROVED** [review]

File: Sources/Sentry/SentryProfiler.mm (hunk 3 above)
Comment (indragiek) [reply Sources/Sentry/SentryProfiler.mm:105]: Will do!
